File: test/bench_ffi.mjs
// Benchmark FFI helpers for the JavaScript target.

// Returns a monotonic timestamp in nanoseconds. `performance.now()` is
// available in Node and modern browsers, returns sub-millisecond
// resolution as a float, and is monotonic across the lifetime of the
// process. Only differences are meaningful.
export function monotonic_ns() {
  return Math.floor(performance.now() * 1_000_000);
}

File: test/bench_ffi.erl
-module(bench_ffi).
-export([monotonic_ns/0]).

%% Returns a monotonic timestamp in nanoseconds. Suitable for measuring
%% elapsed time inside the benchmark harness; the absolute value has no
%% meaning, only differences do.
monotonic_ns() -> erlang:monotonic_time(nanosecond).
